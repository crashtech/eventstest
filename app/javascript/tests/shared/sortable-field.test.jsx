import React from 'react'
import renderer from 'react-test-renderer'

const options = [
  { value: 1, text: 'Option 1' },
  { value: 2, text: 'Option 2' },
  { value: 3, text: 'Option 3' },
]

import SortableField from '../../shared/sortable-field'

test('Can add item from options', () => {
  const mockOnChange = jest.fn()
  const component = renderer.create(<SortableField value={[]} options={[]} onChange={mockOnChange} />)
  expect(component.toJSON()).toMatchSnapshot()

  component.getInstance().addItem(null, { options, value: 2 })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[0][0]).toStrictEqual([2])
})

test('Can add item from any value', () => {
  const mockOnChange = jest.fn()
  const component = renderer.create(<SortableField value={[]} options={[]} onChange={mockOnChange} />)
  expect(component.toJSON()).toMatchSnapshot()

  component.getInstance().addItem(null, { options, value: 'New Value' })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[0][0]).toStrictEqual(['New Value'])
})

test('Can remove item', () => {
  const mockOnChange = jest.fn()
  const component = renderer.create(<SortableField value={[]} options={[]} onChange={mockOnChange} />)
  component.getInstance().addItem(null, { options, value: 1 })
  component.getInstance().addItem(null, { options, value: 2 })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[1][0]).toStrictEqual([1, 2])

  component.getInstance().removeItem(0)
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[2][0]).toStrictEqual([2])
})

test('Can change order of the items', () => {
  const mockOnChange = jest.fn()
  const component = renderer.create(<SortableField value={[]} options={[]} onChange={mockOnChange} />)
  component.getInstance().addItem(null, { options, value: 1 })
  component.getInstance().addItem(null, { options, value: 2 })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[1][0]).toStrictEqual([1, 2])

  component.getInstance().changeOrder({ oldIndex: 0, newIndex: 1 })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
  expect(mockOnChange.mock.calls[2][0]).toStrictEqual([2, 1])
})
