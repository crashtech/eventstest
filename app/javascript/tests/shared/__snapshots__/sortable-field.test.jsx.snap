// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can add item from any value 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  />,
]
`;

exports[`Can add item from any value 2`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      New Value
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Can add item from options 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  />,
]
`;

exports[`Can add item from options 2`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 2
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Can change order of the items 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 1
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 2
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Can change order of the items 2`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 2
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 1
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Can remove item 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 1
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 2
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Can remove item 2`] = `
Array [
  <div
    aria-expanded={false}
    className="ui selection dropdown sortable-field"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <div
      aria-atomic={true}
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    />
  </div>,
  <div
    className="ui divided ordered list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      Option 2
      <i
        aria-hidden="true"
        className="close icon"
        onClick={[Function]}
      />
    </div>
  </div>,
]
`;
